File: examples/with-claims.ts
import { signJwt, verifyJwt } from '../src/index';

const secret = 'claims-secret-change-me';

const token = signJwt(
    { featureFlag: 'beta-access' },
    {
        secret,
        issuer: 'api.my-app.local',
        subject: 'user-1001',
        audience: ['dashboard', 'mobile-app'],
        notBefore: 5,
        expiresIn: 3600
    }
);

const payload = verifyJwt(token, {
    secret,
    issuer: ['api.my-app.local'],
    subject: 'user-1001',
    audience: 'mobile-app',
    clockTolerance: 5,
    allowedClaims: ['featureFlag']
});

console.log('JWT with claims:', token);
console.log('Verified payload:', payload);

try {
    verifyJwt(token, {
        secret,
        allowedClaims: ['scope'] // "featureFlag" is not in the allow list, so this fails
    });
} catch (error) {
    console.log('Strict claim whitelist rejected token:', (error as Error).message);
}

File: examples/with_claims.rs
use hash_token_rust::{
    sign_jwt, verify_jwt, Audience, Issuer, JwtAlgorithm, JwtClaims, SignJwtOptions,
    VerifyJwtOptions,
};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut claims = JwtClaims::new();
    claims.insert("role".to_string(), "admin".into());

    let token = sign_jwt(
        &claims,
        &SignJwtOptions {
            secret: "a-very-secure-secret-value".to_string(),
            algorithm: Some(JwtAlgorithm::HS512),
            expires_in: Some(600.0),
            audience: Some(Audience::Single("internal-api".into())),
            issuer: Some("auth-service".into()),
            subject: Some("user-123".into()),
            ..Default::default()
        },
    )?;

    let verified = verify_jwt(
        &token,
        &VerifyJwtOptions {
            secret: "a-very-secure-secret-value".to_string(),
            algorithms: Some(vec![JwtAlgorithm::HS512]),
            audience: Some(Audience::Single("internal-api".into())),
            issuer: Some(Issuer::Single("auth-service".into())),
            subject: Some("user-123".into()),
            ..Default::default()
        },
    )?;

    println!("claims={verified:?}");
    Ok(())
}
